### PR TITLE
Create pool metadata modal form

### PR DESCRIPTION
**Pull request by devjoaov**
[linear issue](https://linear.app/bleu-llc/issue/BAL-100/implement-pool-metadata-modal-form-component)

rendered component:
image

### CODE REVIEW

File: packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx [hunk 1]
```diff
@@ -173,9 +171,7 @@ export function MetadataAttribute({ poolId }: { poolId: string }) {
         </div>
         <div className="mt-5 w-full justify-between sm:flex sm:items-center">
           <div className="flex gap-4">
-            <Button className="bg-indigo-500 text-white hover:bg-indigo-400 focus-visible:outline-indigo-500">
-              Add new attribute
-            </Button>
+            <PoolMetadataForm />
```
Comment (ribeirojose) [inline packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx:174]: `#would` make it explicit this is a modal:
```suggestion
            <PoolMetadataFormModal />
```
Also it sounds to me like a clearer (and a bit more composable) API passing the the trigger as a child component, so effectively:

```
            <PoolMetadataFormModal>
                <Button className="bg-indigo-500 text-white hover:bg-indigo-400 focus-visible:outline-indigo-500">
                  Add new attribute
                </Button>
            </PoolMetadataFormModal>
```

**Review by ribeirojose — APPROVED** [review]
👍 lgtm

File: packages/next-app/src/components/PoolMetadataForm.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                  disabled={true}
+                  className="bg-indigo-500 text-gray-500 hover:bg-indigo-400 focus-visible:outline-indigo-500 disabled:bg-gray-600"
+                >
+                  Save item
+                </Button>
+              </div>
+            </form>
+          </div>
+          <Dialog.Close asChild>
+            <button
```
[the remark this answers is not in the document]
Comment (devjoaov) [reply packages/next-app/src/components/PoolMetadataForm.tsx:100]: This button is just positioning the "close icon" at the top of the form

image

File: packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx (hunk 1 above)
Comment (devjoaov) [reply packages/next-app/src/app/pool/[poolId]/MetadataAttribute.tsx:174]: That makes sense, thanks sir

**Pull request merged**